apps/notifications/src/panel/templates/button-spam.jsx

```diff
@@ -2,33 +2,36 @@
  * External dependencies
  */
 import PropTypes from 'prop-types';
-
-import React from 'react';
+import React, { useContext } from 'react';
 import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
-import { spamNote } from '../flux/note-actions';
+import { spamNote } from '../state/notes/thunks';
 import ActionButton from './action-button';
 import { keys } from '../helpers/input';
+import { RestClientContext } from '../Notifications';
+
+const SpamButton = ( { note, translate, spamNote } ) => {
+	const restClient = useContext( RestClientContext );
 
-const SpamButton = ( { note, translate } ) => (
-	<ActionButton
-		{ ...{
-			icon: 'spam',
-			isActive: false,
-			hotkey: keys.KEY_S,
-			onToggle: () => spamNote( note ),
-			text: translate( 'Spam', { context: 'verb: Mark as Spam' } ),
-			title: translate( 'Mark comment as spam', { context: 'verb: imperative' } ),
-		} }
-	/>
-);
+	return (
+		<ActionButton
+			icon={ 'spam' }
```

Comment:
Nit: I'd alter that to:
```suggestion
			icon="spam"
```